packages/frontend/src/features/metricsCatalog/components/MetricPeekModal.tsx

```diff
@@ -61,6 +70,43 @@ export const MetricPeekModal: FC<Props> = ({ opened, onClose }) => {
 
     const history = useHistory();
 
+    const currentMetricIndex = useMemo(() => {
+        return metrics.findIndex(
+            (metric) =>
+                metric.name === metricName && metric.tableName === tableName,
+        );
+    }, [metrics, metricName, tableName]);
+
+    const nextMetricInList = useMemo(() => {
+        return metrics[currentMetricIndex + 1];
+    }, [currentMetricIndex, metrics]);
+
+    const previousMetricInList = useMemo(() => {
+        return metrics[currentMetricIndex - 1];
+    }, [currentMetricIndex, metrics]);
+
+    const navigateToMetric = useCallback(
+        (metric: CatalogField) => {
+            history.push({
+                pathname: `/projects/${projectUuid}/metrics/peek/${metric.tableName}/${metric.name}`,
+                search: history.location.search,
+            });
```

Comment:
I think we should also reset state, so if I segment by on metric A, then navigate to metric B, I should see a clean slate - wdyt? Or is this actually a better experience?